presto-spi/src/main/java/com/facebook/presto/spi/block/BlockUtil.java

```diff
@@ -65,7 +65,16 @@ else if (newSize > MAX_ARRAY_SIZE) {
 
     static int calculateBlockResetSize(int currentSize)
     {
-        return intSaturatedCast((long) Math.ceil(currentSize * BLOCK_RESET_SKEW));
+        long newSize = (long) Math.ceil(currentSize * BLOCK_RESET_SKEW);
```

Comment:
Right, but if the operator behaves correctly it shouldn't push much more rows when `maxBlockSizeInBytes` is reached. Probably much less then 25% is required